File: 101-symmetric-tree/101-symmetric-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

bool traverse(struct TreeNode* root1, struct TreeNode* root2){
    
    if(root1==NULL && root2==NULL)
        return true;
    
    else if(root1==NULL || root2==NULL)
        return false;
    
    else if(root1->val != root2->val)
        return false;
    
    return traverse(root1->left,root2->right) && traverse(root1->right,root2->left);    
}

bool isSymmetric(struct TreeNode* root){
    
    if(root==NULL)
        return true;
    
    return traverse(root->left,root->right);

}

File: 101-symmetric-tree/101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
    def traverse(self, left, right):
        
        if(left==None and right==None):
            return True
        
        elif(left==None or right==None):
            return False
        
        elif(left.val!=right.val):
            return False

        
        return self.traverse(left.left,right.right) and self.traverse(left.right,right.left) 
        
    
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        
        return self.traverse(root.left,root.right)